addons/mail/static/src/messaging/entity/partner/partner.js

```diff
@@ -0,0 +1,250 @@
+            await this._fetchImStatus();
+            this._loopFetchImStatus();
+        }
+
+        async checkIsUser() {
+            const userIds = await this.env.rpc({
+                model: 'res.users',
+                method: 'search',
+                args: [[['partner_id', '=', this.id]]],
+            });
+            this.update({ userId: userIds.length ? userIds[0] : null });
+        }
+
+        /**
+         * @returns {string}
+         */
+        get nameOrDisplayName() {
+            return this.name || this.display_name;
+        }
+
+        /**
+         * Opens an existing or new chat.
+         */
+        openChat() {
+            const chat = this.directPartnerThread;
+            if (chat) {
+                chat.open();
+            } else {
+                this.env.entities.Thread.createChannel({
+                    autoselect: true,
+                    partnerId: this.id,
+                    type: 'chat',
+                });
+            }
+        }
+
+        //----------------------------------------------------------------------
+        // Private
+        //----------------------------------------------------------------------
+
+        /**
+         * @static
+         * @private
+         * @param {Object} param0
+         * @param {Object} param0.env
+         */
+        static async _fetchImStatus() {
+            let toFetchPartnersLocalIds = [];
+            let partnerIdToLocalId = {};
+            const toFetchPartners = this.all.filter(partner => partner.im_status !== null);
+            for (const partner of toFetchPartners) {
+                toFetchPartnersLocalIds.push(partner.localId);
+                partnerIdToLocalId[partner.id] = partner.localId;
+            }
+            if (!toFetchPartnersLocalIds.length) {
+                return;
+            }
+            const dataList = await this.env.rpc({
+                route: '/longpolling/im_status',
+                params: {
+                    partner_ids: toFetchPartnersLocalIds.map(partnerLocalId =>
+                        this.get(partnerLocalId).id
+                    ),
+                },
+            }, { shadow: true });
+            for (const { id, im_status } of dataList) {
+                this.insert({ id, im_status });
+                delete partnerIdToLocalId[id];
+            }
+            // partners with no im_status => set null
+            for (const noImStatusPartnerLocalId of Object.values(partnerIdToLocalId)) {
+                const partner = this.get(noImStatusPartnerLocalId);
+                if (partner) {
+                    partner.update({ im_status: null });
+                }
+            }
+        }
+
+        /**
+         * @static
+         * @private
+         */
+        static _loopFetchImStatus() {
+            setTimeout(async () => {
+                await this._fetchImStatus();
+                this._loopFetchImStatus();
+            }, 50*1000);
```

Comment:
```suggestion
            }, 50 * 1000);
```
And probably better to have that time as a class variable.